### PR TITLE
workerd: new package

### PR DESCRIPTION
**Pull request by isker**
[workerd](https://github.com/cloudflare/workerd) is yet another JavaScript runtime for the server, like [node](https://github.com/wolfi-dev/os/blob/main/nodejs-20.yaml), [deno](https://github.com/wolfi-dev/os/blob/main/deno.yaml), etc.

This PR is incomplete, the build fails with obscure linker problems, related to [workerd's static linkage against libc++](https://github.com/cloudflare/workerd#building-workerd).  I'm posting it because I'm at the limit of my abilities to fight native build tooling 🌞.  Feel free to close it at any time.

This package builds against the libc++ package recently added in https://github.com/wolfi-dev/os/pull/4852.

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

**Comment by isker** [comment]
Here are the linker errors I see when running in melange locally:

```
$ melange build --arch x86_64 workerd.yaml --cache-dir /tmp/melange-cache

. . .

⚠️  x86_64    | ERROR: /var/cache/melange/bazel/_bazel_build/3eb1481cb157e75aa22f6b17db8bf7a1/external/capnp-cpp/src/capnp/BUILD.bazel:125:10: Linking external/capnp-cpp/src/capnp/capnp_tool [for tool] failed: (Exit 1): clang-16 failed: error executing command (from target @capnp-cpp//src/capnp:capnp_tool) 
⚠️  x86_64    |   (cd /var/cache/melange/bazel/_bazel_build/3eb1481cb157e75aa22f6b17db8bf7a1/sandbox/linux-sandbox/6/execroot/workerd && \
⚠️  x86_64    |   exec env - \
⚠️  x86_64    |     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
⚠️  x86_64    |     PWD=/proc/self/cwd \
⚠️  x86_64    |   /usr/bin/clang-16 @bazel-out/k8-opt-exec-2B5CBBC6/bin/external/capnp-cpp/src/capnp/capnp_tool-2.params)
⚠️  x86_64    | # Configuration: a5832f54de26a3679d3536042872962f043ba3740fb29b893bf0deb8df071b01
⚠️  x86_64    | # Execution platform: @local_config_platform//:host
⚠️  x86_64    | 
⚠️  x86_64    | Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
⚠️  x86_64    | ld.lld: error: undefined symbol: vtable for std::runtime_error
⚠️  x86_64    | >>> referenced by stdexcept.cpp
⚠️  x86_64    | >>>               stdexcept.cpp.o:(std::runtime_error::runtime_error(char const*)) in archive /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.2.0/../../../../lib64/libc++.a
⚠️  x86_64    | >>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)
⚠️  x86_64    | 
⚠️  x86_64    | ld.lld: error: undefined symbol: operator new(unsigned long)
⚠️  x86_64    | >>> referenced by stdexcept.cpp
⚠️  x86_64    | >>>               stdexcept.cpp.o:(std::runtime_error::runtime_error(char const*)) in archive /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.2.0/../../../../lib64/libc++.a
⚠️  x86_64    | >>> referenced by mutex.c++
⚠️  x86_64    | >>>               bazel-out/k8-opt-exec-2B5CBBC6/bin/external/capnp-cpp/src/kj/_objs/kj/mutex.o:(kj::_::Mutex::checkPredicate(kj::_::Mutex::Waiter&))
⚠️  x86_64    | >>> referenced by schema-loader.c++
⚠️  x86_64    | >>>               bazel-out/k8-opt-exec-2B5CBBC6/bin/external/capnp-cpp/src/capnp/_objs/capnp/schema-loader.o:(capnp::SchemaLoader::SchemaLoader(capnp::SchemaLoader::LazyLoadCallback const&))
⚠️  x86_64    | >>> referenced 130 more times
```

It goes on and on like that.  I don't know if some of the [unusual flags](https://github.com/wolfi-dev/os/blob/16e1781fd5b44825e9be3db4897d2f0e30fde922/llvm-libcxx-16.yaml#L42-L44) being passed in libc++'s build are to blame, or something else.

File: workerd.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    with:
+      repository: https://github.com/cloudflare/workerd
+      tag: v${{package.version}}
+      expected-commit: b4b25357f47e737621098dde66fa0d0c6cb58450
+      destination: workerd
+
+  - runs: |
+      ln -s /var/cache/melange .cache
+      cd workerd
+      JAVA_HOME=/usr/lib/jvm/java-11-openjdk bazel sync --configure
```
Comment (isker) [inline workerd.yaml:47]: The `ln` and `bazel sync` are only useful for local development (to persist the bazel cache across builds) and should be deleted before merging.

Bazel inside of melange is a bit of a pain.

**Comment by rawlingsj** [comment]
Is this PR still needed?  Looks like it's become stale.

**Comment by isker** [comment]
Yes, I gave up.

**Comment by sschepens** [comment]
@isker i created #8689 for fixing `libcxx` build flags, and I've already got merged `libcxx-15` which has the correct build flags and am using it for building envoy statically linked with `libc++`. Maybe you can give `llvm-libcxx-15` a try for this PR? until we get `16` and `17` fixed

**Comment by isker** [comment]
Well done, and thanks for letting me know.  I don’t think I’ll be making another attempt.

**Pull request closed without merging**